### PR TITLE
[Dev] `STANDARD_VECTOR_SIZE` and `BLOCK_ALLOC_SIZE` can now be set through the Makefile

### PR DESCRIPTION
**Pull request by Tishj**
Title kind of says it all

- `STANDARD_VECTOR_SIZE` translates to the `STANDARD_VECTOR_SIZE` define in `vector_size.hpp`
- `BLOCK_ALLOC_SIZE` translates to the (new) `DUCKDB_BLOCK_ALLOC_SIZE` define in `storage_info.hpp`, this is then used to set `Storage::BLOCK_ALLOC_SIZE`.

I looked at the `test_block_sizes.py` script for the `BLOCK_ALLOC_SIZE`
`BLOCK_ALLOC_SIZE` was by default set to `DEFAULT_BLOCK_ALLOC_SIZE`

The script only replaces that one instance, but `DEFAULT_BLOCK_ALLOC_SIZE` is also referenced in `src/storage/single_file_block_manager.cpp`, I'm not sure if that should also be changed?

### CODE REVIEW

**Comment by Tishj** [comment]
As a follow up, we can rework the vector size and block size tests we have that rely on manually editing the files
I've committed those changes by accident more than once, I figured this was cleaner and less error prone

**Comment by taniabogatsch** [comment]
Cool! Can we trigger the nightly runs on this? I want to make sure that this does not break the block size nightly.

**Review by taniabogatsch — COMMENTED** [review]
I just had a look at this. Currently, you cannot compile with any `BLOCK_ALLOC_SIZE`. Have a look at the changes in `feature`. There is a minimum block allocation size, and a test that ensures the block allocation size is a power of two (and maybe others, there is a verify function in `storage_info`). Maybe we should add these here, similar to this.

```cpp
#if (STANDARD_ROW_GROUPS_SIZE < STANDARD_VECTOR_SIZE)
#error Row groups must be able to hold at least one vector
#endif
```

**Comment by Tishj** [comment]
Yea there are requirements, the compilation will fail if those are not met, I ran into those when I tried a random `BLOCK_ALLOC_SIZE`, as it needs to be a multiple of `Storage::SECTOR_SIZE`.

`STANDARD_VECTOR_SIZE` has a similar requirements, needing to be a power of two (though that doesn't seem to be entirely enforced, because 12 was allowed(?))

Those are respected even when set this way

**Comment by taniabogatsch** [comment]
Ah, great!

**Review by carlopi — CHANGES_REQUESTED** [review]
I think also changing/removing scripts/test_block_sizes.py and scripts/test_vector_sizes.py to use standard Make / CMake instead of replacements would be cool, and allow to test this PR.

That said, it's already an improvement as is.

Changes to nightly-tests have to be reverted (I think), so the "request changes".

File: .github/workflows/NightlyTests.yml [hunk 1]
```diff
@@ -14,7 +14,6 @@ on:
   pull_request:
     types: [opened, reopened, ready_for_review]
     paths-ignore:
-      - '**'
```
[the remark this answers is not in the document]
Comment (Tishj) [reply .github/workflows/NightlyTests.yml:17]: > Can we trigger the nightly runs on this?

Yes I'm aware 👍

[the remark this answers is not in the document]
Comment (carlopi) [reply .github/workflows/NightlyTests.yml:17]: I though so, we need a mechanism that matches "override current logic, trust me, and do run test X on this PR". But that's not for today.

**Comment by Tishj** [comment]
I changed the `test_block_sizes.py` and `test_vector_sizes.py` files, seeing as I have to make another commit anyways to revert the nightly test stuff

I'll push it after the relevant test finishes to show that everything works

**Comment by Tishj** [comment]
@taniabogatsch it didn't complete successfully, but I'm not sure the failures are related to these changes?
<https://github.com/duckdb/duckdb/actions/runs/9177094312/job/25234470190?pr=12164>

**Comment by Tishj** [comment]
Seems to pass the nightly tests:
<https://github.com/Tishj/duckdb/actions/runs/9188051251/job/25267556702>
<https://github.com/Tishj/duckdb/actions/runs/9188051251/job/25267558237>

**Review by carlopi — APPROVED** [review]

**Review by taniabogatsch — APPROVED** [review]

**Comment by taniabogatsch** [comment]
Could you also add this? We have some places where we implicitly assume this, I know you said earlier that it will fail. But I'd rather have it explicitly in the `storage_info.hpp`, where we declare them.
```cpp
#if (DUCKDB_BLOCK_ALLOC_SIZE & (DUCKDB_BLOCK_ALLOC_SIZE - 1) != 0)
#error The duckdb block allocation size must be a power of two
#endif
```

Also not sure if this will fail if it exceeds the maximum value. As it could still be a power of two. But we have some places where we just assume this maximum when casting. I've added more checks [here](https://github.com/duckdb/duckdb/pull/12235), but it would be nice to include this also for `DUCKDB_BLOCK_ALLOC_SIZE`.
```cpp
#if (DUCKDB_BLOCK_ALLOC_SIZE > 2147483647)
#error The duckdb block allocation size must not exceed the maximum value of a 32-bit signed integer
#endif
```

And maybe also this?
```cpp
#if (DUCKDB_BLOCK_ALLOC_SIZE < MIN_BLOCK_ALLOC_SIZE)
#error The duckdb block allocation size must be greater or equal than the minimum block allocation size
#endif
```

**Comment by Mytherin** [comment]
Thanks! Great idea

**Pull request merged**